test/js_test.go

```diff
@@ -2875,6 +2876,53 @@ func testJetStreamMirror_Source(t *testing.T, nodes ...*jsServer) {
 		if got != expectedTotal {
 			t.Errorf("Expected %v, got: %v", expectedTotal, got)
 		}
+
+		got = len(si.Sources)
+		expected := 2
+		if got != expected {
+			t.Errorf("Expected %v, got: %v", expected, got)
+		}
+	})
+
+	t.Run("update stream with sources", func(t *testing.T) {
+		si, err := js.StreamInfo("s1")
+		if err != nil {
+			t.Fatalf("Unexpected error creating stream: %v", err)
+		}
+		got := len(si.Config.Sources)
+		expected := 2
+		if got != expected {
+			t.Errorf("Expected %v, got: %v", expected, got)
+		}
+
+		got = len(si.Sources)
+		if got != expected {
+			t.Errorf("Expected %v, got: %v", expected, got)
+		}
+
+		// Make an update
+		config := si.Config
+		config.MaxMsgs = 128
+		updated, err := js.UpdateStream(&config)
+		if err != nil {
+			t.Fatalf("Unexpected error creating stream: %v", err)
+		}
+
+		got = len(updated.Config.Sources)
+		if got != expected {
+			t.Errorf("Expected %v, got: %v", expected, got)
+		}
+
+		got = len(updated.Sources)
+		if got != expected {
+			t.Errorf("Expected %v, got: %v", expected, got)
```

Comment:
On previous revision of the server I was getting 0 here, seems fixed after upgrading to latest in the go.mod:
```
js_test.go:2918: Expected 2, got: 0
```